reloc.ml

```diff
@@ -251,7 +251,16 @@ let quote_files cmdline lst =
 let cygpath l cont =
   let accept_error = (!use_cygpath = `Try && l <> []) in
   let l =
-    let args = String.concat " " (List.map Filename.quote l) in
+    let s = String.concat " " (List.map Filename.quote l) in
+    let args =
+      if String.length s >= 1024 then begin
```

Comment:
Cygwin isn't affected by command line length (different mechanism used), so out of an abundance of caution, can we have:
```suggestion
      if not Sys.cygwin && String.length s >= 1024 then begin
```

Reply:
Looks like `Sys.cygwin` is only `4.01+`.

Reply:
I changed the check to `Sys.os_type <> "Cygwin"`

Reply:
I've now added a compatibility shim. Thanks @MisterDA !